internal/pkg/service/common/etcdop/op/op_result.go

```diff
@@ -4,10 +4,16 @@ import (
 	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
 )
 
+// Result of a high-level operation.
 type Result[R any] struct {
+	*resultBase
+	result *R
+}
+
+// resultBase is common part of all results regardless of the type of value or whether it is a transaction.
+type resultBase struct {
```

Comment:
Improved code about operations results, op and txn results are now compatible, and duplicate methods can be removed.